avae/evaluate.py

```diff
@@ -40,16 +42,21 @@ def evaluate(datapath, lim, splt, batch_s, collect_meta, use_gpu):
     # ############################### MODEL ###############################
     device = set_device(use_gpu)
 
-    if not os.path.exists("states"):
-        raise RuntimeError(
-            "There are no existing model states saved, unable to evaluate."
-        )
-    # TODO add param to chose model
-    states = sorted([s for s in os.listdir("states") if ".pt" in s])[0]
-    fname = states.split(".")[0].split("_")
+    if state is None:
+        if not os.path.exists("states"):
+            raise RuntimeError(
+                "There are no existing model states saved or provided via the state flag in config unable to evaluate."
+            )
+        else:
+            state = sorted([s for s in os.listdir("states") if ".pt" in s])[0]
+            state = os.path.join("states", state)
+
+    fname = state.split(".")[0].split("_")
     pose_dims = fname[3]
-    vae = torch.load(os.path.join("states", states))  # make optional param
 
+    vae = []
+    checkpoint = torch.load(state)
+    vae.load_state_dict(checkpoint["model_state_dict"])
```

Comment:
i'm not sure what is this supposed to do, but it breaks the code.

Reply:
vae is the class and not a list.

Reply:
Camila to fix

Reply:
Fixed in [38415ec](https://github.com/alan-turing-institute/affinity-vae/pull/127/commits/38415ecc1bce0811ff5a0fddc03915e18ef396a1)